Process YAML frontmatter in .md files

redoPop opened the pull request:
Allows markdown files to be treated as doc blocks if they open with YAML.

* [Example of a markdown file with YAML frontmatter](https://github.com/redoPop/hologram/blob/md-docblock/spec/fixtures/source/components/bem.md) – note that GitHub [has special support for rendering YAML frontmatter in markdown](https://github.com/blog/1647-viewing-yaml-metadata-in-your-documents).
* [Example of HTML output](https://github.com/redoPop/hologram/blob/md-docblock/spec/fixtures/styleguide/code.html) generated from the `bem.md` source in the previous example. (These files are used for for new tests.)
* Compatibility is maintained for markdown files that don't include YAML; e.g. [the `index.md` fixture](https://github.com/redoPop/hologram/blob/md-docblock/spec/fixtures/source/components/index.md). Plain markdown files are treated just as they were before.

Summary of change:

`process_files` now examines the contents of each `.md` file to see if its opening lines look like YAML. If so, the file's contents are added to the `doc_block_collection`.

Resolves #188 since title can be specified in the YAML configuration.

My own use case is producing a CSS _code_ style guide alongside the pattern style guide – i.e., documentation that doesn't really belong in a Sass/scss file. Using `.md` files for this documentation also makes them readable directly in GitHub, which is a nice bonus.

jdcantrell commented:
Sorry for the slow response this is really nice. I'm wondering if the yaml shouldn't have a key specific in the yaml to let hologram know it should parse the yaml and markdown.

jdcantrell commented:
The idea being that you might have yaml+markdown for generating something else in your project and you wouldn't want hologram to try and create pages off it.

redoPop commented:
Can do! How does this sound:

    ---
    hologram: true
    title:    (...etc.)
    ---
    
    (Markdown body)

* If there is YAML frontmatter in the markdown file _and_ that frontmatter includes `hologram: true`, then it is processed by Hologram as a doc block.
* Otherwise Hologram falls back to its default handling of markdown files (adding them as pages with an `:md` key to be handled by the default renderer).

jdcantrell commented:
I think it's good, I was trying to think of something else but that's probably the best option for now. Thanks again for the PR.

redoPop commented:
Okidoke! I also added an additional fixture for a dummy "jekyll.md" file that doesn't meet the `hologram: true` test, and added info about the new key to the readme.

An idea: we could avoid the need for the `hologram: true` flag by looking for a combination of the keys `title`, `name`, and one of `parent`/`category`/`categories` in the YAML. That footprint seems fairly unique; if the markdown contains all three perhaps we can be reasonably confident we're looking at a Hologram configuration.
